src/extension/infer.rs

```diff
@@ -1092,4 +1104,183 @@ mod test {
         }
         Ok(())
     }
+
+    fn make_opaque(extension: impl Into<ExtensionId>, signature: FunctionType) -> ops::LeafOp {
+        let opaque =
+            ops::custom::OpaqueOp::new(extension.into(), "", "".into(), vec![], Some(signature));
+        ops::custom::ExternalOp::from(opaque).into()
+    }
+
+    /// A CFG rooted hugr adding resources at each basic block.
+    /// Looks like this:
+    ///
+    ///          +-------------+
+    ///          |    Entry    |
+    ///          |  (Adds [A]) |
+    ///          +-/---------\-+
+    ///           /           \
+    ///  +-------/-----+     +-\-------------+
+    ///  |     BB0     |     |      BB1      |
+    ///  | (Adds [BC]) |     |   (Adds [B])  |
+    ///  +----\--------+     +---/------\----+
+    ///        \                /        \
+    ///         \              /          \
+    ///          \       +----/-------+  +-\---------+
```

Comment:
I think you could make this a slightly more interesting test by linking BB0 to BB10 (rather than to exit)....but,

I think you could make this a *lot* more interesting by linking, say, `BB10`, *back* to `BB1` or, even better, `Entry` to make a loop. I guess the hope is that unification will deal with loops no problem, without requiring iteration to a fixpoint (say), but it'd be great to have the test of that :)